### PR TITLE
BAEL-223 - Integrate DynamoDB into a Spring Boot Application using Sp…

### PR DESCRIPTION
**Pull request by anataliocs**
Integrating DynamoDB Into a Spring Boot Application Using Spring Data

For Issue:  http://jira.baeldung.com/browse/BAEL-223

http://inprogress.baeldung.com/?p=18138&preview=true

### CODE REVIEW

File: spring-boot-data-dynamodb/src/main/java/com/baeldung/spring/data/es/config/DynamoDBConfig.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+    @Value("${amazon.aws.secretkey}")
+    private String amazonAWSSecretKey;
+
+    @Bean
+    public AmazonDynamoDB amazonDynamoDB() {
+        AmazonDynamoDB amazonDynamoDB = new AmazonDynamoDBClient(amazonAWSCredentials());
+        if (StringUtils.isEmpty(amazonDynamoDBEndpoint)) {
+            amazonDynamoDB.setEndpoint(amazonDynamoDBEndpoint);
+        }
```
Comment (pkhanal) [inline spring-boot-data-dynamodb/src/main/java/com/baeldung/spring/data/es/config/DynamoDBConfig.java]: Should be **StringUtils.isNotEmpty**
It does not make sense to set the endpoint when it is empty :)
You might got confused when you referred to the sample from https://github.com/derjust/spring-data-dynamodb

File: spring-boot-data-dynamodb/src/test/java/com/baeldung/spring/data/es/repository/ProductInfoRepositoryIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+    @Autowired
+    ProductInfoRepository ProductInfoRepository;
+
+    private static final String EXPECTED_COST = "20";
+    private static final String EXPECTED_PRICE = "50";
+
+    @Before
+    @Ignore
+    public void setUp() throws Exception {
```
Comment (pkhanal) [inline spring-boot-data-dynamodb/src/test/java/com/baeldung/spring/data/es/repository/ProductInfoRepositoryIntegrationTest.java]: You have ```@BeforeClass``` in the article, however you have ```@Before``` here. Please make it consistent. Based on the feedback, I think it should be ```@BeforeClass```. And the method needs to be ```static``` when we use ```@BeforeClass```

Comment (anataliocs) [inline spring-boot-data-dynamodb/src/test/java/com/baeldung/spring/data/es/repository/ProductInfoRepositoryIntegrationTest.java]: Updated

File: spring-boot-data-dynamodb/src/main/java/com/baeldung/spring/data/es/config/DynamoDBConfig.java (hunk 1 above)
Comment (anataliocs) [inline spring-boot-data-dynamodb/src/main/java/com/baeldung/spring/data/es/config/DynamoDBConfig.java]: This has been updated.  The org.springframework.util.StringUtils does not have a isNotEmpty method so I just changed it to NOT isEmpty

File: spring-boot-data-dynamodb/README.MD [hunk 3]
```diff
@@ -0,0 +1,2 @@
+###The Course
+The "REST With Spring" Classes: http://bit.ly/restwithspring
```
Comment (pkhanal) [inline spring-boot-data-dynamodb/README.MD]: Why do we have this link in the README? May be it is not relevant here.

File: spring-boot-data-dynamodb/src/main/resources/application.properties [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+management.address=127.0.0.1
+
+endpoints.shutdown.enabled=true
+
+endpoints.jmx.domain=Spring Sample Application
+endpoints.jmx.uniqueNames=true
+
+##jolokia.config.debug=true
+##endpoints.jolokia.enabled=true
+##endpoints.jolokia.path=jolokia
```
Comment (pkhanal) [inline spring-boot-data-dynamodb/src/main/resources/application.properties]: Let's cleanup these unused and commented out properties

File: spring-boot-data-dynamodb/src/main/resources/logback.xml [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+			<pattern>web - %date [%thread] %-5level %logger{36} - %message%n
+			</pattern>
+		</encoder>
+	</appender>
+
+	<root level="INFO">
+		<appender-ref ref="STDOUT" />
+	</root>
+
+</configuration>
```
Comment (pkhanal) [inline spring-boot-data-dynamodb/src/main/resources/logback.xml]: We have indentation issue here. Check out how it is done is other projects. Usually **4 spaces** preferred over **tab**. It's a minor issue but it is good to consistent with other projects.

File: spring-boot-data-dynamodb/src/main/resources/templates/index.html [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+		  <a href="#" class="close" data-dismiss="alert" aria-label="close">�</a>
+		  <strong>Success!</strong> It is working as we expected.
+		</div>
+	</div>
+
+<script src="/webjars/jquery/2.1.4/jquery.min.js"></script>
+<script src="/webjars/bootstrap/3.3.4/js/bootstrap.min.js"></script>
+
+</body>
+</html>
```
Comment (pkhanal) [inline spring-boot-data-dynamodb/src/main/resources/templates/index.html]: Similar comment regarding indentation issue.

File: spring-boot-data-dynamodb/README.MD (hunk 3 above)
Comment (anataliocs) [inline spring-boot-data-dynamodb/README.MD]: updated @pkhanal

File: spring-boot-data-dynamodb/src/main/resources/application.properties (hunk 4 above)
Comment (anataliocs) [inline spring-boot-data-dynamodb/src/main/resources/application.properties]: updated @pkhanal

File: spring-boot-data-dynamodb/src/main/resources/logback.xml (hunk 5 above)
Comment (anataliocs) [inline spring-boot-data-dynamodb/src/main/resources/logback.xml]: image

It is 4 spaces in the actual code if you open it up.  It just is formatted to look like a tab in the github UI

Comment (anataliocs) [inline spring-boot-data-dynamodb/src/main/resources/logback.xml]: image

It is 4 spaces.  The github UI makes it look like a tab.  @pkhanal

File: spring-boot-data-dynamodb/src/main/resources/templates/index.html (hunk 6 above)
Comment (anataliocs) [inline spring-boot-data-dynamodb/src/main/resources/templates/index.html]: image

It is 4 spaces. The github UI makes it look like a tab. @pkhanal

**Comment by anataliocs** [comment]
@pkhanal I have updated the PR as per your feedback

File: spring-boot-data-dynamodb/src/main/resources/logback.xml (hunk 5 above)
Comment (anataliocs) [inline spring-boot-data-dynamodb/src/main/resources/logback.xml]: Also @pkhanal this project was copied from an existing accepted project and was not changed

File: spring-boot-data-dynamodb/src/main/resources/templates/index.html (hunk 6 above)
Comment (anataliocs) [inline spring-boot-data-dynamodb/src/main/resources/templates/index.html]: Also this project was copied from an existing accepted project and was not changed @pkhanal

File: spring-boot-data-dynamodb/src/test/java/com/baeldung/spring/data/es/repository/ProductInfoRepositoryIntegrationTest.java [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+
+    @Autowired
+    ProductInfoRepository ProductInfoRepository;
+
+    private static final String EXPECTED_COST = "20";
+    private static final String EXPECTED_PRICE = "50";
+
+    @BeforeClass
+    @Ignore
+    public static void setUp() throws Exception {
```
Comment (pkhanal) [inline spring-boot-data-dynamodb/src/test/java/com/baeldung/spring/data/es/repository/ProductInfoRepositoryIntegrationTest.java]: rename to **setup**

File: spring-boot-data-dynamodb/src/test/java/com/baeldung/spring/data/es/repository/ProductInfoRepositoryIntegrationTest.java [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+
+        try {
+            dynamoDBMapper = new DynamoDBMapper(amazonDynamoDB);
```
Comment (pkhanal) [inline spring-boot-data-dynamodb/src/test/java/com/baeldung/spring/data/es/repository/ProductInfoRepositoryIntegrationTest.java]: I am afraid you cannot access non static field inside the *static* method. Please fix the compilation error.

File: spring-boot-data-dynamodb/src/test/java/com/baeldung/spring/data/es/repository/ProductInfoRepositoryIntegrationTest.java [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+    @Autowired
+    private AmazonDynamoDB amazonDynamoDB;
+
+    @Autowired
+    ProductInfoRepository ProductInfoRepository;
+
+    private static final String EXPECTED_COST = "20";
+    private static final String EXPECTED_PRICE = "50";
+
+    @BeforeClass
```
Comment (pkhanal) [inline spring-boot-data-dynamodb/src/test/java/com/baeldung/spring/data/es/repository/ProductInfoRepositoryIntegrationTest.java]: You will need to import the *BeforeClass*

File: spring-boot-data-dynamodb/pom.xml [hunk 10]
```diff
@@ -0,0 +1,185 @@
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+	<modelVersion>4.0.0</modelVersion>
+	<groupId>com.baeldung</groupId>
+	<artifactId>spring-boot</artifactId>
```
Comment (pkhanal) [inline spring-boot-data-dynamodb/pom.xml]: Change the artifactId to **spring-boot-dynamodb**.

**Comment by anataliocs** [comment]
Updated all requests and also cleaned up code and updated dependencies

Integration test runs successfully
image

mvn clean install runs successfully
image

**Pull request merged**